### DISCUSSION TITLE
Android编译失败

### DISCUSSION THREAD

**Discussion by member-zhang** [alibaba/xquic#41] [Q&A]
[ 91%] Linking CXX executable crypto_test
ld: error: undefined symbol: abi_test_trampoline
>>> referenced by abi_self_test.cc:44 (/mnt/data/xquic/third_party/boringssl/crypto/abi_self_test.cc:44)
>>>               CMakeFiles/crypto_test.dir/abi_self_test.cc.o:(ABITest_SanityCheck_Test::TestBody())
>>> referenced by abi_self_test.cc:44 (/mnt/data/xquic/third_party/boringssl/crypto/abi_self_test.cc:44)
>>>               CMakeFiles/crypto_test.dir/abi_self_test.cc.o:(ABITest_SanityCheck_Test::TestBody())
>>> referenced by abi_self_test.cc:425 (/mnt/data/xquic/third_party/boringssl/crypto/abi_self_test.cc:425)
>>>               CMakeFiles/crypto_test.dir/abi_self_test.cc.o:(ABITest_AArch64_Test::TestBody())
>>> referenced 2 more times

ld: error: undefined symbol: abi_test_clobber_x19
>>> referenced by abi_self_test.cc:425 (/mnt/data/xquic/third_party/boringssl/crypto/abi_self_test.cc:425)
>>>               CMakeFiles/crypto_test.dir/abi_self_test.cc.o:(ABITest_AArch64_Test::TestBody())
>>> referenced by abi_self_test.cc:425 (/mnt/data/xquic/third_party/boringssl/crypto/abi_self_test.cc:425)
>>>               CMakeFiles/crypto_test.dir/abi_self_test.cc.o:(ABITest_AArch64_Test::TestBody())

ld: error: undefined symbol: abi_test_clobber_x0
>>> referenced by abi_self_test.cc:429 (/mnt/data/xquic/third_party/boringssl/crypto/abi_self_test.cc:429)
>>>               CMakeFiles/crypto_test.dir/abi_self_test.cc.o:(ABITest_AArch64_Test::TestBody())
>>> referenced by abi_self_test.cc:429 (/mnt/data/xquic/third_party/boringssl/crypto/abi_self_test.cc:429)
>>>               CMakeFiles/crypto_test.dir/abi_self_test.cc.o:(ABITest_AArch64_Test::TestBody())

ld: error: undefined symbol: abi_test_clobber_x1
>>> referenced by abi_self_test.cc:430 (/mnt/data/xquic/third_party/boringssl/crypto/abi_self_test.cc:430)
>>>               CMakeFiles/crypto_test.dir/abi_self_test.cc.o:(ABITest_AArch64_Test::TestBody())
>>> referenced by abi_self_test.cc:430 (/mnt/data/xquic/third_party/boringssl/crypto/abi_self_test.cc:430)
>>>               CMakeFiles/crypto_test.dir/abi_self_test.cc.o:(ABITest_AArch64_Test::TestBody())

ld: error: undefined symbol: abi_test_clobber_x2
>>> referenced by abi_self_test.cc:431 (/mnt/data/xquic/third_party/boringssl/crypto/abi_self_test.cc:431)
>>>               CMakeFiles/crypto_test.dir/abi_self_test.cc.o:(ABITest_AArch64_Test::TestBody())
>>> referenced by abi_self_test.cc:431 (/mnt/data/xquic/third_party/boringssl/crypto/abi_self_test.cc:431)
>>>               CMakeFiles/crypto_test.dir/abi_self_test.cc.o:(ABITest_AArch64_Test::TestBody())

ld: error: undefined symbol: abi_test_clobber_x3
>>> referenced by abi_self_test.cc:432 (/mnt/data/xquic/third_party/boringssl/crypto/abi_self_test.cc:432)
>>>               CMakeFiles/crypto_test.dir/abi_self_test.cc.o:(ABITest_AArch64_Test::TestBody())
>>> referenced by abi_self_test.cc:432 (/mnt/data/xquic/third_party/boringssl/crypto/abi_self_test.cc:432)
>>>               CMakeFiles/crypto_test.dir/abi_self_test.cc.o:(ABITest_AArch64_Test::TestBody())

ld: error: undefined symbol: abi_test_clobber_x4
>>> referenced by abi_self_test.cc:433 (/mnt/data/xquic/third_party/boringssl/crypto/abi_self_test.cc:433)
>>>               CMakeFiles/crypto_test.dir/abi_self_test.cc.o:(ABITest_AArch64_Test::TestBody())
>>> referenced by abi_self_test.cc:433 (/mnt/data/xquic/third_party/boringssl/crypto/abi_self_test.cc:433)
>>>               CMakeFiles/crypto_test.dir/abi_self_test.cc.o:(ABITest_AArch64_Test::TestBody())

ld: error: undefined symbol: abi_test_clobber_x5
>>> referenced by abi_self_test.cc:434 (/mnt/data/xquic/third_party/boringssl/crypto/abi_self_test.cc:434)
>>>               CMakeFiles/crypto_test.dir/abi_self_test.cc.o:(ABITest_AArch64_Test::TestBody())
>>> referenced by abi_self_test.cc:434 (/mnt/data/xquic/third_party/boringssl/crypto/abi_self_test.cc:434)
>>>               CMakeFiles/crypto_test.dir/abi_self_test.cc.o:(ABITest_AArch64_Test::TestBody())

ld: error: undefined symbol: abi_test_clobber_x6
>>> referenced by abi_self_test.cc:435 (/mnt/data/xquic/third_party/boringssl/crypto/abi_self_test.cc:435)
>>>               CMakeFiles/crypto_test.dir/abi_self_test.cc.o:(ABITest_AArch64_Test::TestBody())
>>> referenced by abi_self_test.cc:435 (/mnt/data/xquic/third_party/boringssl/crypto/abi_self_test.cc:435)
>>>               CMakeFiles/crypto_test.dir/abi_self_test.cc.o:(ABITest_AArch64_Test::TestBody())

ld: error: undefined symbol: abi_test_clobber_x7
>>> referenced by abi_self_test.cc:436 (/mnt/data/xquic/third_party/boringssl/crypto/abi_self_test.cc:436)
>>>               CMakeFiles/crypto_test.dir/abi_self_test.cc.o:(ABITest_AArch64_Test::TestBody())
>>> referenced by abi_self_test.cc:436 (/mnt/data/xquic/third_party/boringssl/crypto/abi_self_test.cc:436)
>>>               CMakeFiles/crypto_test.dir/abi_self_test.cc.o:(ABITest_AArch64_Test::TestBody())

ld: error: undefined symbol: abi_test_clobber_x8
>>> referenced by abi_self_test.cc:437 (/mnt/data/xquic/third_party/boringssl/crypto/abi_self_test.cc:437)
>>>               CMakeFiles/crypto_test.dir/abi_self_test.cc.o:(ABITest_AArch64_Test::TestBody())
>>> referenced by abi_self_test.cc:437 (/mnt/data/xquic/third_party/boringssl/crypto/abi_self_test.cc:437)
>>>               CMakeFiles/crypto_test.dir/abi_self_test.cc.o:(ABITest_AArch64_Test::TestBody())

ld: error: undefined symbol: abi_test_clobber_x9
>>> referenced by abi_self_test.cc:438 (/mnt/data/xquic/third_party/boringssl/crypto/abi_self_test.cc:438)
>>>               CMakeFiles/crypto_test.dir/abi_self_test.cc.o:(ABITest_AArch64_Test::TestBody())
>>> referenced by abi_self_test.cc:438 (/mnt/data/xquic/third_party/boringssl/crypto/abi_self_test.cc:438)
>>>               CMakeFiles/crypto_test.dir/abi_self_test.cc.o:(ABITest_AArch64_Test::TestBody())

ld: error: undefined symbol: abi_test_clobber_x10
>>> referenced by abi_self_test.cc:439 (/mnt/data/xquic/third_party/boringssl/crypto/abi_self_test.cc:439)
>>>               CMakeFiles/crypto_test.dir/abi_self_test.cc.o:(ABITest_AArch64_Test::TestBody())
>>> referenced by abi_self_test.cc:439 (/mnt/data/xquic/third_party/boringssl/crypto/abi_self_test.cc:439)
>>>               CMakeFiles/crypto_test.dir/abi_self_test.cc.o:(ABITest_AArch64_Test::TestBody())

ld: error: undefined symbol: abi_test_clobber_x11
>>> referenced by abi_self_test.cc:440 (/mnt/data/xquic/third_party/boringssl/crypto/abi_self_test.cc:440)
>>>               CMakeFiles/crypto_test.dir/abi_self_test.cc.o:(ABITest_AArch64_Test::TestBody())
>>> referenced by abi_self_test.cc:440 (/mnt/data/xquic/third_party/boringssl/crypto/abi_self_test.cc:440)
>>>               CMakeFiles/crypto_test.dir/abi_self_test.cc.o:(ABITest_AArch64_Test::TestBody())

ld: error: undefined symbol: abi_test_clobber_x12
>>> referenced by abi_self_test.cc:441 (/mnt/data/xquic/third_party/boringssl/crypto/abi_self_test.cc:441)
>>>               CMakeFiles/crypto_test.dir/abi_self_test.cc.o:(ABITest_AArch64_Test::TestBody())
>>> referenced by abi_self_test.cc:441 (/mnt/data/xquic/third_party/boringssl/crypto/abi_self_test.cc:441)
>>>               CMakeFiles/crypto_test.dir/abi_self_test.cc.o:(ABITest_AArch64_Test::TestBody())

ld: error: undefined symbol: abi_test_clobber_x13
>>> referenced by abi_self_test.cc:442 (/mnt/data/xquic/third_party/boringssl/crypto/abi_self_test.cc:442)
>>>               CMakeFiles/crypto_test.dir/abi_self_test.cc.o:(ABITest_AArch64_Test::TestBody())
>>> referenced by abi_self_test.cc:442 (/mnt/data/xquic/third_party/boringssl/crypto/abi_self_test.cc:442)
>>>               CMakeFiles/crypto_test.dir/abi_self_test.cc.o:(ABITest_AArch64_Test::TestBody())

ld: error: undefined symbol: abi_test_clobber_x14
>>> referenced by abi_self_test.cc:443 (/mnt/data/xquic/third_party/boringssl/crypto/abi_self_test.cc:443)
>>>               CMakeFiles/crypto_test.dir/abi_self_test.cc.o:(ABITest_AArch64_Test::TestBody())
>>> referenced by abi_self_test.cc:443 (/mnt/data/xquic/third_party/boringssl/crypto/abi_self_test.cc:443)
>>>               CMakeFiles/crypto_test.dir/abi_self_test.cc.o:(ABITest_AArch64_Test::TestBody())

ld: error: undefined symbol: abi_test_clobber_x15
>>> referenced by abi_self_test.cc:444 (/mnt/data/xquic/third_party/boringssl/crypto/abi_self_test.cc:444)
>>>               CMakeFiles/crypto_test.dir/abi_self_test.cc.o:(ABITest_AArch64_Test::TestBody())
>>> referenced by abi_self_test.cc:444 (/mnt/data/xquic/third_party/boringssl/crypto/abi_self_test.cc:444)
>>>               CMakeFiles/crypto_test.dir/abi_self_test.cc.o:(ABITest_AArch64_Test::TestBody())

ld: error: undefined symbol: abi_test_clobber_x16
>>> referenced by abi_self_test.cc:445 (/mnt/data/xquic/third_party/boringssl/crypto/abi_self_test.cc:445)
>>>               CMakeFiles/crypto_test.dir/abi_self_test.cc.o:(ABITest_AArch64_Test::TestBody())
>>> referenced by abi_self_test.cc:445 (/mnt/data/xquic/third_party/boringssl/crypto/abi_self_test.cc:445)
>>>               CMakeFiles/crypto_test.dir/abi_self_test.cc.o:(ABITest_AArch64_Test::TestBody())

ld: error: undefined symbol: abi_test_clobber_x17
>>> referenced by abi_self_test.cc:446 (/mnt/data/xquic/third_party/boringssl/crypto/abi_self_test.cc:446)
>>>               CMakeFiles/crypto_test.dir/abi_self_test.cc.o:(ABITest_AArch64_Test::TestBody())
>>> referenced by abi_self_test.cc:446 (/mnt/data/xquic/third_party/boringssl/crypto/abi_self_test.cc:446)
>>>               CMakeFiles/crypto_test.dir/abi_self_test.cc.o:(ABITest_AArch64_Test::TestBody())

ld: error: too many errors emitted, stopping now (use -error-limit=0 to see all errors)
clang++: error: linker command failed with exit code 1 (use -v to see invocation)
third_party/boringssl/crypto/CMakeFiles/crypto_test.dir/build.make:1713: recipe for target 'third_party/boringssl/crypto/crypto_test' failed
make[2]: *** [third_party/boringssl/crypto/crypto_test] Error 1
CMakeFiles/Makefile2:422: recipe for target 'third_party/boringssl/crypto/CMakeFiles/crypto_test.dir/all' failed
make[1]: *** [third_party/boringssl/crypto/CMakeFiles/crypto_test.dir/all] Error 2
Makefile:83: recipe for target 'all' failed
make: *** [all] Error 2

本机的环境是Ubuntu18.04，ndk版本是22.1.7171670

**Comment by yuanbo-zhang** [comment]
看起来是boringssl测试相关编译失败了，单独编译boringssl你环境上能编译通过吗
